### PR TITLE
Adding TLS and SMTP Authentication

### PR DESCRIPTION
**Pull request by muonzoo**
Here's a clean pull request.  Fixed earlier problem with smtp_host.

### CODE REVIEW

File: lib/urlwatch/mailer.py [hunk 1]
```diff
@@ -1,15 +1,48 @@
...
+                'keyring for {}, {}'.format(smtp_server, from_email))
+        s.login(from_email, passwd)
     s.sendmail(from_email, [to_email], msg.as_string())
     s.quit()
 
+
+def set_password(smtp_server, from_email):
+    ''' Set the keyring password for the mail connection. Interactive.'''
+    if keyring is None:
+        raise ImportError('keyring module missing - service unspported')
```
Comment (thp) [inline lib/urlwatch/mailer.py]: Typo: unspported -> unsupported

File: urlwatch [hunk 2]
```diff
@@ -181,12 +184,18 @@ if __name__ == '__main__':
         log.info('turning display of errors ON')
         display_errors = True
 
+    if options.password:
+        mailer.set_password(options.email_smtp,options.email_from)
```
Comment (thp) [inline urlwatch]: PEP-8 nitpick: space after `,`

File: urlwatch [hunk 3]
```diff
@@ -383,7 +393,8 @@ if __name__ == '__main__':
                 subject = 'Changes detected (%d)' % len(summary)
                 mailer.send(email_smtp_server, email_sender_address,
                         email_receiver_address, subject,
-                        short_summary + '\n' + '\n'.join(details))
+                            short_summary + '\n' + '\n'.join(details),
+                            auth=email_auth,tls=email_tls)
```
Comment (thp) [inline urlwatch]: Also here, space after `,`

**Comment by thp** [comment]
Added some minor comments, after fixing them (e.g. with `git commit --amend` to update the PR) and a force-push to your branch, I think it's ready for merge.

**Comment by muonzoo** [comment]
Thanks - I ran this through pep8 and didn't get the same results - is there a preferred tool that you use to find these quickly? I'm happy to incorporate it into my regular development cycle.

**Comment by muonzoo** [comment]
Fixed as requested.

**Pull request merged**